app/src/main/java/org/xbmc/kore/host/HostInfo.java

```diff
@@ -55,11 +54,20 @@
 	 */
 	public static final int DEFAULT_EVENT_SERVER_PORT = 9777;
 
-    public static final int DEFAULT_KODI_VERSION_MAJOR = 16;
+	public static final int KODI_V12_FRODO = 12;
```

Comment:
Why did you include FRODO, HELIX, ISENGARD, and LEIA? They're not used. Is it just for completeness?

Reply:
I'm confused why you use Kodi version at all, isn't the JSON API version what matters?

Reply:
i'd agree.
Every Kodi release there's a JSON  version bump done. Either a minor or major version.

Reply:
@poisdeux Yes, it's for completeness (though i didn't bother to implement the rest of `isXXXorHigher` functions...)

@DaveTBlake @MartijnKaijser 
Kind of. First, even though this PR includes changes to these constants, this particular change doesn't discriminate based on Kodi versions or JSON RPC API versions: It's a change of behaviour (getting the genres from songs instead of albums) to all versions of Kodi.

We currently discriminate behaviour based on Kodi versions on 2 situations:
1. To open the subtitles search, which was an Addon pre-Gotham and a internal window post-Gotham. I'm not sure, but this change in Kodi might not have been reflected in a JSON RPC API bump, as it has nothing to do with it;
2. To display codec info, whose window changed from Kripton onwards *in event server*.

AFAICT, these situations have anything to do with JSON RPC API changes, so we use Kodi's version to change the behaviour.

Reply:
But i must say that JSON RPC versioning is something that isn't well handled by Kore right now (it's handled on a case by case basis, without any support infrastructure for versioning). We haven't had any problems so far, but we might have to look at it more thoroughly .